msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFFileExportMethod.java

```diff
@@ -0,0 +1,205 @@
+  private boolean canceled = false;
+  private int totalScans = 0;
+  private float progress = 0f;
+  private int lastLoggedProgress = 0;
+
+  public NetCDFFileExportMethod(@Nonnull RawDataFile rawDataFile, @Nonnull File target) {
+    this(rawDataFile, target, 1, 1);
+  }
+
+  public NetCDFFileExportMethod(@Nonnull RawDataFile rawDataFile, @Nonnull File target,
+      double massValueScaleFactor, double intensityValueScaleFactor) {
+    this.rawDataFile = rawDataFile;
+    this.target = target;
+    this.massValueScaleFactor = massValueScaleFactor;
+    this.intensityValueScaleFactor = intensityValueScaleFactor;
+  }
+
+  @Override
+  public Void execute() throws MSDKException {
+
+    logger.info("Started export of " + rawDataFile.getName() + " to " + target);
+
+    List<MsScan> scans = rawDataFile.getScans();
+    totalScans = scans.size();
+    int scanStartPositions[] = new int[totalScans + 1];
+
+    try {
+
+      NetcdfFileWriter writer =
+          NetcdfFileWriter.createNew(Version.netcdf3, target.getAbsolutePath());
+
+      // Populate the scan indices
+      // Create a simple 1D array of type int, element i of the shape array contains the length of
+      // the (i+1)th dimension of the array
+      Array scanIndexArray = Array.factory(int.class, new int[] {totalScans});
+      int idx = 0;
+      for (MsScan scan : scans) {
+
+        if (canceled) {
+          writer.close();
+          return null;
+        }
+
+        scanStartPositions[idx + 1] = scanStartPositions[idx] + scan.getNumberOfDataPoints();
+        idx++;
+
+      }
+
+      for (int i = 0; i < scanStartPositions.length - 1; i++)
+        scanIndexArray.setInt(i, scanStartPositions[i]);
+
+      // Write the scan indices
+      // Create a variable and set the dimension scan_number=totalScans to it
+      Dimension scanNumberDim = writer.addDimension(null, "scan_number", totalScans);
+      List<Dimension> scanIndexDims = new ArrayList<>();
+      scanIndexDims.add(scanNumberDim);
+      Variable scanIndexVariable =
+          writer.addVariable(null, "scan_index", DataType.INT, scanIndexDims);
+
+      progress = 0.2f;
+
+      // data values storage dimension
+      // Dimension to store the data arrays of various scans
+      Dimension pointNumDim =
+          writer.addDimension(null, "point_number", scanStartPositions[totalScans]);
+      List<Dimension> pointNumValDims = new ArrayList<>();
+      pointNumValDims.add(pointNumDim);
+
+      // populate the mass and the intensity values
+      Array massValueArray =
+          Array.factory(double.class, new int[] {scanStartPositions[totalScans]});
+      Array intensityValueArray =
+          Array.factory(double.class, new int[] {scanStartPositions[totalScans]});
+
+      idx = 0;
+      double mzValues[] = new double[10000];
+      float intensityValues[] = new float[10000];
+      for (MsScan scan : scans) {
+        mzValues = scan.getMzValues();
```

Comment:
This is kind of missing the the point, it has to be
````
mzValues = scan.getMzValues(mzValues);
````
otherwise the arrays are not being reused